compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -1371,4 +1418,22 @@ object ParallelTesting {
 
   def isTastyFile(f: JFile): Boolean =
     f.getName.endsWith(".tasty")
+
+  def getCompiler(version: String): JFile =
+    val dir = cache.resolve(s"scala3-${version}").toFile
+    synchronized {
+      if dir.exists then
+        dir
+      else
+        import scala.sys.process._
+        val zipPath = cache.resolve(s"scala3-$version.tar.gz")
+        val compilerDownloadUrl = s"https://github.com/lampepfl/dotty/releases/download/$version/scala3-$version.tar.gz"
```

Comment:
Downloading the same files every time the CI is run isn't great: it will both slow down the CI and cost EPFL money (we never got into troubles but they do keep track of bandwidth usage). To avoid this we have permanent caches for dependencies: https://github.com/lampepfl/dotty/blob/8d28d94009236026a4b51437b0246a463b3d61f3/.github/workflows/ci.yaml#L33-L36 It'd be great if you could either use `~/.cache` or  rely on coursier (or scala-cli ?) to fetch a specific version from maven.

Reply:
OK, I'll try to optimize that somehow